promql/engine.go

```diff
@@ -1488,6 +1510,7 @@ func (ev *evaluator) matrixIterSlice(it *storage.BufferedSeriesIterator, mint, m
 	buf := it.Buffer()
 	for buf.Next() {
 		t, v := buf.At()
+		ev.samplesStats.Increment(1)
```

Comment:
This is going to under count, as a given sample might be processed in multiple steps.

You'll probably need to track this over around line 1150 instead.